### ISSUE TITLE
macOS Sequoia 15.2 Update Broke Dark Mode

### ISSUE THREAD

**Issue by delacrixmorgan** [git-up/GitUp#1042]
Ever since I've updated to Sequoia 15.2 on GitUp 1.4.2 (1051), the red branch UI is broken in dark mode. I've wondered if anyone faced the same issue.

| Light Mode ☀️ | Dark Mode 🌚 |
|---------------|--------------|
| <img width="852" alt="Screenshot 2024-12-17 at 10 22 28 AM" src="https://github.com/user-attachments/assets/30ee601a-0baa-4b96-bdd7-89692c16b7b4" /> |  <img width="875" alt="Screenshot 2024-12-17 at 10 22 10 AM" src="https://github.com/user-attachments/assets/73a14133-5af4-469f-ad61-4c1372987c5e" /> |

**Comment by ruslankmlv** [comment]
I'm having the same issue after the update to 15.2 as well

**Comment by lucasderraugh** [comment]
Dang, so it's the upgrade that suddenly cause it. I had this only in 1 repo in particular and so I figured it was something related to https://github.com/git-up/GitUp/issues/336 but the fact that switching to light mode reveals all is very weird.

Thanks for flagging this, I'll try to get some time this week to look at it as it's also driving me a bit up the wall.

**Comment by lucasderraugh** [comment]
Looks like if I remove the conditional on this and always multiply it works as expected:
https://github.com/git-up/GitUp/blob/2a2db81c323d2befbb2f42fba1a27fae77e5b9b5/GitUpKit/Interface/GIGraphView.m#L1536-L1538

I don't really understand why though.

**Comment by richardtop** [comment]
<img width="370" alt="Screenshot 2024-12-31 at 5 29 35" src="https://github.com/user-attachments/assets/8608693f-ffaf-4b33-a85b-0ce7b4d6f43c" />
<img width="118" alt="Screenshot 2024-12-31 at 5 28 59" src="https://github.com/user-attachments/assets/852f0a00-f6e8-476c-b5db-5d2ede446c5e" />

**Comment by kangu** [comment]
I have the same issue as well. It looks to be somehow intermittent though... it only happens on what seem to be large repositories with many commits (around 3000 for me) and the first time it loads, it looks ok, but the moment the focus state is activated on one of the nodes, the branch line disappears. I'm on GitUp 1.4.2 and Sequoia 15.2

**Comment by lucasderraugh** [comment]
@kangu I agree. I have some repos that work fine and others that don't. Given that I can't find any real common thread I may just through in a build that has my change above that appears to resolve the issue for reasons I don't understand. It's definitely a pretty obnoxious issue.

**Comment by mluisbrown** [comment]
I have the same issue and so far it's affecting all my repos. It's not a blocker though, I can still use GitUp fine with this issue, it just doesn't look as nice.

**Comment by stodirascu** [comment]
No worries, thanks @lucasderraugh, looking forward

**Comment by lucasderraugh** [comment]
Ya, I was hoping to find some other approach here, but no luck other than changing the blend modes. This has the less than ideal change of making the intersections of the lines quite dark in dark mode colors and looks worse than before IMO.

However, I'd rather have lines drawing than not at all. I was hoping there was something I was missing in the logic but I couldn't find anything obvious and it's much less obvious why changing the blend mode to multiply (and specifically multiply) should suddenly make the lines appear.

I should probably file a feedback with Apple on this, but my time on this project is limited as it is.
I'll roll a new continuous build tomorrow, I've merged the "fix" on master in the meantime. I'll close this ticket once the build is officially out.

**Comment by stodirascu** [comment]
Ok thanks, looking forward for the build and appreciate you looking into it. And agree that slightly uglier intersections are better than invisible lines, until a better fix is to be found.

These "system update breaks current app" issues are the most joyful.

**Comment by lucasderraugh** [comment]
Alright, I guess the answer is to never work from an airport 😅. The continuous channel should now have this fixed, generated the wrong signature somehow.

**Comment by stodirascu** [comment]
"Never work from an airport" - that should be in the release notes 😄

Thanks a bunch!
